### PR TITLE
🏃 Add a modified util function

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

**What this PR does / why we need it**:
This PR is needed to separate concerns of structs in the KubeadmControlPlane reconciler.

This function that is copied in this PR does too much by requiring a client and actually creating the objects. There is already something that is responsible for communicating with the API server so this function does not need to do that.

This is the code that is being modified: https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/external/util.go#L17

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #2702

### CODE REVIEW

**Comment by vincepri** [comment]
Could we instead have a different intermediate function in the other package that does exactly this with less input?

**Comment by chuckha** [comment]
> Could we instead have a different intermediate function in the other package that does exactly this with less input?

Absolutely fine with me, lemme refator

**Comment by chuckha** [comment]
very slightly awkward as to not change the existing API.

File: controllers/external/util.go [hunk 1]
```diff
@@ -81,11 +81,55 @@ func CloneTemplate(ctx context.Context, in *CloneTemplateInput) (*corev1.ObjectR
...
+	// OwnerRef is an optional OwnerReference to attach to the cloned object.
+	// +optional
+	OwnerRef *metav1.OwnerReference
+
+	// Labels is an optional map of labels to be added to the object.
+	// +optional
+	Labels map[string]string
+}
+
+func GenerateTemplate(in GenerateTemplateInput) (*unstructured.Unstructured, error) {
```
Comment (vincepri) [inline controllers/external/util.go]:
```suggestion
func GenerateTemplate(in *GenerateTemplateInput) (*unstructured.Unstructured, error) {
```

for consistency with the other function

Comment (chuckha) [reply controllers/external/util.go]: fixed!

**Pull request merged**